File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        int i = 0;
        int j = 0;

        while (i < str1.size() && j < str2.size()) {

            if (str1[i] == str2[j] || (str1[i] + 1) == str2[j] || str2[j] == 'a' && str1[i] == 'z'){
                j ++;
            }

            i ++;
            
        }
        return j == str2.size();
    }
};

File: 3018-make-string-a-subsequence-using-cyclic-increments/3018-make-string-a-subsequence-using-cyclic-increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        i = 0

        for c in str1:
            if i < len(str2) and (ord(str2[i]) - ord(c) + 26) % 26 <= 1:
                i += 1

        return i == len(str2)